src/lib/registration/connect_helpers.rb

```diff
@@ -63,6 +63,7 @@ class ConnectHelpers
     def self.catch_registration_errors(message_prefix: "",
       show_update_hint: false,
       silent_reg_code_mismatch: false,
+      retry_block: false,
```

Comment:
Do we need an extra parameter to enable it? I mean, in my opinion it makes sense always retrying...

Reply:
I was not sure if was fine in all the scenarios, so added it as an option (probably if silent_reg_code_mismatch is true then the retry should be skipped).

If it would be nice to have it always, then yes, the parameter is not needed.

Reply:
`silent_reg_code_mismatch` is for automatic trying the base product reg. code also for addons (if fails we ask the user for the reg code). But if it times out here then I think we should report it to the user as it is very likely that it would timeout also with the user provided reg. code.

Reply:
To make it short: retry always, user can always press "cancel" if it is not desired.